### PR TITLE
Use duration instead of claim end in storage

### PR DESCRIPTION
**Pull request by fedgiac**
Addresses two comments from the audit, namely:
- by storing `duration` instead of `end` you can save a little bit of calculation
- it's probably better to convert `start` and `end` to `uint256` than to convert `block.timestamp` to `uint32` [here]( https://github.com/cowprotocol/team-cow-allocation/blob/63f03106615352e9fb52abe6e52f159552d6ca60/src/contracts/AllocationModule.sol#L217)

The end result is simpler code and a small gas cost improvement (~100 gas/call).
Saturating to `uint32` is not needed anymore as we don't need to cast to `uint32`.

A small difference from the current code is that now we can create valid claims that expire after the timestamp represented by the max value for a `uint32` (the longest claim is one starting at `uint32(-1)` with duration `uint32(-1)`, which would expire in the year 2242).

### Test Plan

Existing unit tests. Note that we already test that `claimCow [...] can be redeemed if current timestamp does not fit a uint32`.

### CODE REVIEW

File: src/contracts/AllocationModule.sol [hunk 1]
```diff
@@ -19,8 +19,8 @@ contract AllocationModule {
         uint96 claimedAmount;
         /// @dev Timestamp when this vesting position started.
         uint32 start;
-        /// @dev Timestamp when this vesting position ends.
-        uint32 end;
+        /// @dev How long after vesting start must be waited before claiming the total amount.
```
Comment (josojo) [inline src/contracts/AllocationModule.sol:22]: Maybe: Timespan between vesting start and ending

**Review by nlordell — APPROVED** [review]
LGTM

**Review by josojo — APPROVED** [review]
Good catch from Adam. The gas savings would not be worth it, but I like that the code is simpler now.

**Review by bh2smith — APPROVED** [review]
Very interesting change!

**Pull request merged**